src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Parse user's input, identify whether it is a valid command and calls relevant method(s) to execute the command.
+ */
+
+public class Parser {
+    private final TaskList tasks;
+
+    /**
+     * The constructor for a Parser Object.
+     */
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * This method checks if a String starting with delete is a delete command.
+     * A delete command is a string with the keyword delete followed by numbers and spaces only.
+     * Examples of valid user inputs:
+     * -> delete 3
+     * -> delete   4(spaces trimmed)
+     * -> delete 1 4(spaces trimmed and interpret as done 14)
+     * Examples of invalid user inputs:
+     * -> delete task
+     * -> delete 2 task
```

Comment:
Description of the method should be simple. Additional comments on the code can be done in the readme.md